OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalCacheCleaner.java

```diff
@@ -61,73 +66,13 @@ public void run() {
-                        OneSignalDbContract.InAppMessageTable.COLUMN_NAME_MESSAGE_ID,
-                        OneSignalDbContract.InAppMessageTable.COLUMN_CLICK_IDS
-                };
-
-                String whereStr = OneSignalDbContract.InAppMessageTable.COLUMN_NAME_LAST_DISPLAY + " < ?";
-
-                String sixMonthsAgoInSeconds = String.valueOf((System.currentTimeMillis() / 1_000L) - IAM_CACHE_DATA_LIFETIME);
-                String[] whereArgs = new String[]{sixMonthsAgoInSeconds};
-
-                Set<String> oldMessageIds = OSUtils.newConcurrentSet();
-                Set<String> oldClickedClickIds = OSUtils.newConcurrentSet();
-
-                Cursor cursor = null;
-                try {
-                    cursor = writableDb.query(OneSignalDbContract.InAppMessageTable.TABLE_NAME,
-                            retColumns,
-                            whereStr,
-                            whereArgs,
-                            null,
-                            null,
-                            null);
-
-                    if (cursor == null || cursor.getCount() == 0) {
-                        OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Attempted to clean 6 month old IAM data, but none exists!");
-                        return;
-                    }
-
-                    // From cursor get all of the old message ids and old clicked click ids
-                    if (cursor.moveToFirst()) {
-                        do {
-                            String oldMessageId = cursor.getString(
-                                    cursor.getColumnIndex(
-                                            OneSignalDbContract.InAppMessageTable.COLUMN_NAME_MESSAGE_ID));
-                            String oldClickIds = cursor.getString(
-                                    cursor.getColumnIndex(
-                                            OneSignalDbContract.InAppMessageTable.COLUMN_CLICK_IDS));
-
-                            oldMessageIds.add(oldMessageId);
-                            oldClickedClickIds.addAll(OSUtils.newStringSetFromJSONArray(new JSONArray(oldClickIds)));
-                        } while (cursor.moveToNext());
-                    }
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                } finally {
-                    if (cursor != null && !cursor.isClosed())
-                        cursor.close();
-                }
-
-                // 2. Delete old IAMs from SQL
-                writableDb.delete(
-                        OneSignalDbContract.InAppMessageTable.TABLE_NAME,
-                        whereStr,
-                        whereArgs);
-
-                // 3. Use queried data to clean SharedPreferences
-                cleanInAppMessageIds(oldMessageIds);
-                cleanInAppMessageClickedClickIds(oldClickedClickIds);
+                new OSInAppMessageRepository(dbHelper).cleanCachedInAppMessages();
```

Comment:
maybe access it throw OneSignal.java? or add it to the method params? because there should be already a repository instance alive if we have more that one instance then te synchronized won't work